### PR TITLE
[13.0][MIG] stock_archive_constraint: Migration to 13.0

### PR DESCRIPTION
**Pull request by victoralmau**
Allows to block archiving products and locations with associated `stock.quant` or `stock.move`

Related to: https://github.com/OCA/stock-logistics-warehouse/pull/1019 (12.0)

Some changes have been made to adapt it to 13.0

Please, @pedrobaeza and @sergio-teruel review it.
@Tecnativa TT25465

### CODE REVIEW

**Review by sergio-teruel — APPROVED** [review]
Code review

**Comment by pedrobaeza** [comment]
Why only one commit? It should contain the whole history for also easing the review and focus on changes from 12.0 to 13.0, as I have already reviewed 12.0 version.

**Comment by victoralmau** [comment]
Ok, sorry, try to redo it again and separate in 2 commits.

File: stock_archive_constraint/models/product_product.py [hunk 1]
```diff
@@ -10,43 +10,52 @@ class ProductProduct(models.Model):
 
     @api.constrains("active")
     def _check_active_stock_archive_constraint_stock_quant(self):
-        res = self.env['stock.quant'].search(
+        res = self.env["stock.quant"].search(
```
Comment (pedrobaeza) [inline stock_archive_constraint/models/product_product.py:13]: This is pre-commit stuff, so it shouldn't be also in the migration to v13 commit. You should follow full migration guide.

Comment (victoralmau) [reply stock_archive_constraint/models/product_product.py:13]: It's necessary to merge PR in 12.0 previously to use migration guideline.

Comment (pedrobaeza) [reply stock_archive_constraint/models/product_product.py:13]: Not really, you just need to follow the steps about pre-commit stuff and commit apart. That's what I want.

**Comment by kos94ok-3D** [comment]
It is not a new module, so set `MIG` tag instead of `ADD`

File: stock_archive_constraint/README.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+#. Go to Settings,> Users> Edit a user and check the "Manage Multiple Stock Locations" permission
```
Comment (CarlosRoca13) [inline stock_archive_constraint/README.rst]:
```suggestion
#. Go to Settings > Users > Edit a user and check the "Manage Multiple Stock Locations" permission
```

**Review by CarlosRoca13 — APPROVED** [review]
LGTM :+1:  Just a minor comment

**Pull request merged**